openvm/src/customize_exe.rs

```diff
@@ -241,72 +241,74 @@ pub fn customize(
 
     let extensions = apcs
         .into_iter()
-        .map(
-            |(
-                Apc {
-                    block,
-                    opcode,
-                    machine,
-                    subs,
+        .enumerate()
+        .map(|(i, (apc, apc_stats))| {
+            let Apc {
+                block,
+                machine,
+                subs,
+            } = apc;
+            let opcode = POWDR_OPCODE + i;
+            // Create a new instruction that will be used to replace the original instructions in the block.
+            // Note that this instruction is never actually looked up, because our APCs do not contain any
+            // PC lookup (instead, they hard-code a PC in the execution bridge receive).
+            // Replacing the instruction here has the effect that the prover is forced to use the APC.
+            // We could also skip this to allow the prover to take either the software or APC path.
```

Comment:
I think the main argument against doing this now is that it complicates witgen, because it is no longer deterministic which executor to run.